src/MLIAP/mliap_so3.cpp

```diff
@@ -0,0 +1,1422 @@
+using namespace jacobi_pd;
+#endif
+
+using namespace LAMMPS_NS;
+using namespace MathConst;
+using namespace MathSpecial;
+
+MLIAP_SO3::MLIAP_SO3(LAMMPS *lmp, double vrcut, int vlmax, int vnmax,
+                     double valpha) : Pointers(lmp)
+{
+  m_rcut = vrcut;
+  m_alpha = valpha;
+  m_lmax = vlmax;
+  m_nmax = vnmax;
+  compute_ncoeff();
+
+  m_Nmax = (m_nmax + m_lmax + 1) * 10;
+
+  m_ellpl1 = nullptr;
+  m_ellm1 = nullptr;
+  m_pfac = nullptr;
+  m_dfac0 = nullptr;
+  m_dfac1 = nullptr;
+  m_dfac2 = nullptr;
+  m_dfac3 = nullptr;
+  m_dfac4 = nullptr;
+  m_dfac5 = nullptr;
+
+  m_g_array = nullptr;
+  m_w = nullptr;
+  m_rootpq = nullptr;
+  m_idxu_block = nullptr;
+  m_idxylm = nullptr;
+
+  m_rip_array = nullptr;
+  m_rip_darray = nullptr;
+
+  m_sbes_array = nullptr;
+  m_sbes_darray = nullptr;
+
+  m_plist_r = nullptr;
+  m_plist_i = nullptr;
+
+  m_clist_r = nullptr;
+  m_clist_i = nullptr;
+
+  m_ulist_r = nullptr;
+  m_ulist_i = nullptr;
+
+  m_Ylms_r = nullptr;
+  m_Ylms_i = nullptr;
+
+  m_dYlm_r = nullptr;
+  m_dYlm_i = nullptr;
+
+  m_dplist_r = nullptr;
+  m_dplist_i = nullptr;
+
+  m_dclist_r = nullptr;
+  m_dclist_i = nullptr;
+
+  m_tempdp_r = nullptr;
+
+  m_clisttot_r = nullptr;
+  m_clisttot_i = nullptr;
+
+  m_init_arrays = 0;
+
+}
+
+MLIAP_SO3::~MLIAP_SO3()
+{
+
+  memory->destroy(m_ellpl1);
+  memory->destroy(m_ellm1);
+  memory->destroy(m_pfac);
+  memory->destroy(m_dfac0);
+  memory->destroy(m_pfac);
+  memory->destroy(m_dfac0);
+  memory->destroy(m_dfac1);
+  memory->destroy(m_dfac2);
+  memory->destroy(m_dfac3);
+  memory->destroy(m_dfac4);
+  memory->destroy(m_dfac5);
+  memory->destroy(m_w);
+  memory->destroy(m_g_array);
+
+  memory->destroy(m_rootpq);
+  memory->destroy(m_idxu_block);
+  memory->destroy(m_idxylm);
+
+  memory->destroy(m_rip_array);
+  memory->destroy(m_rip_darray);
+
+  memory->destroy(m_sbes_array);
+  memory->destroy(m_sbes_darray);
+
+  memory->destroy(m_plist_r);
+  memory->destroy(m_plist_i);
+
+  memory->destroy(m_clist_r);
+  memory->destroy(m_clist_i);
+
+  memory->destroy(m_ulist_r);
+  memory->destroy(m_ulist_i);
+
+  memory->destroy(m_Ylms_r);
+  memory->destroy(m_Ylms_i);
+  memory->destroy(m_dYlm_r);
+  memory->destroy(m_dYlm_i);
+
+  memory->destroy(m_dplist_r);
+  memory->destroy(m_dplist_i);
+
+  memory->destroy(m_dclist_r);
+  memory->destroy(m_dclist_i);
+
+  memory->destroy(m_tempdp_r);
+
+  memory->destroy(m_clisttot_r);
+  memory->destroy(m_clisttot_i);
+
+}
+
+void MLIAP_SO3::compute_ncoeff()
+{
+
+  ncoeff = m_nmax * (m_nmax + 1) * (m_lmax + 1) / 2;
+
+}
+
+void MLIAP_SO3::init()
+{
+  int i, totali;
+
+  totali = m_lmax + 1;
+  memory->destroy(m_ellpl1);
+  memory->create(m_ellpl1, totali, "MLIAP_SO3:m_ellpl1");
+  memory->destroy(m_ellm1);
+  memory->create(m_ellm1, totali, "MLIAP_SO3:m_ellm1");
+
+  for (int l = 1; l < m_lmax + 1; l++) {
+    m_ellpl1[l] = get_sum(0, l + 2, 1, 2);
+    m_ellm1[l] = get_sum(0, l, 1, 2);
+  }
+
+  double pfac1 = 1.0 / 4.0 / M_PI;
```

Comment:
the `M_PI` constant is a commonly used but non-standard extension to the C/C++ standard. Using it unconditionally will cause compilation failure on some platforms. Please use `MY_PI` from the `MathConst` namespace. That will also promote a platform independent value for PI.